test/package/subscriberPackageVersionInstall.test.ts

```diff
@@ -216,7 +220,7 @@ describe('Package Install', () => {
     expect(toolingCreateStub.calledOnce).to.be.true;
 
     // verify we polled
-    expect(retrieveStub.calledThrice).to.be.true;
+    expect(retrieveStub.callCount).to.be.equal(4);
     expect(lifecycleStub.callCount).to.equal(3);
     expect(lifecycleStub.args[0][0]).to.equal('Package/install-presend');
```

Comment:
I'd strongly encourage getting the calls and args, do a flatMap, then asser what you need to be included in them.

This pattern is very brittle (some new warning or telemetry gets added and it breaks tests) 

Even where order matters, you could verify that A comes before B without asserting that they're exactly 0 and 1.